### ISSUE TITLE
Add @JsonFallback annotation for Property Constructor to handle Unknown Fields.

### ISSUE THREAD

**Issue by huhlig** [FasterXML/jackson-annotations#147]
Request to create a new Annotation called @JsonFallback to handle any unknown fields in a property based Constructor. This allows for the normal usage of a property constructor and strong typing but also allows for final classes to be constructed with "additional" unbound fields. This could be useful with a String/Object, String/String Map or similar.

```java
public class MyClass {
    public final int id;
    public final String name;
    public final Map<String,Object>extraFields;
    public MyClass(
        @JsonProperty("id") final int id,
        @JsonProperty("name") final String name;
        @JsonFallback Map<String, Object>extraFields
    ) {
        this.id = id;
        this.name = name;
        this.extraFields = ImmutableMap.of(extraFields);
    }
}```

**Comment by cowtowncoder** [comment]
Usage makes sense, but I think there is already an issue in `jackson-databind` where `@JsonAnySetter` annotation would serve the same role:

https://github.com/FasterXML/jackson-databind/issues/562

which I think might do what you want? (if it gets implemented)

**Comment by cowtowncoder** [comment]
@huhlig Fair point, but I think chances are better than age would suggest mostly because:

1. We are working on 3.0 (... although, with time constraints etc, slowly...)
2. I consider this both useful and more important (relatively speaking) than many other things.

But someone still needs to work on it: it's not trivial to implement: not necessarily difficult, but lots of work to pipe things through.

**Comment by huhlig** [comment]
Alright, I've closed this out and commented on the other ticket.
